fuel-tx/src/transaction/consensus_parameters.rs

```diff
@@ -80,69 +88,90 @@ impl ConsensusParameters {
     pub const fn tx_params(&self) -> &TxParameters {
         match self {
             Self::V1(params) => &params.tx_params,
+            Self::V2(params) => &params.tx_params,
         }
     }
 
     /// Get the predicate parameters
     pub const fn predicate_params(&self) -> &PredicateParameters {
         match self {
             Self::V1(params) => &params.predicate_params,
+            Self::V2(params) => &params.predicate_params,
         }
     }
 
     /// Get the script parameters
     pub const fn script_params(&self) -> &ScriptParameters {
         match self {
             Self::V1(params) => &params.script_params,
+            Self::V2(params) => &params.script_params,
         }
     }
 
     /// Get the contract parameters
     pub const fn contract_params(&self) -> &ContractParameters {
         match self {
             Self::V1(params) => &params.contract_params,
+            Self::V2(params) => &params.contract_params,
         }
     }
 
     /// Get the fee parameters
     pub const fn fee_params(&self) -> &FeeParameters {
         match self {
             Self::V1(params) => &params.fee_params,
+            Self::V2(params) => &params.fee_params,
         }
     }
 
     /// Get the chain ID
     pub const fn chain_id(&self) -> ChainId {
         match self {
             Self::V1(params) => params.chain_id,
+            Self::V2(params) => params.chain_id,
         }
     }
 
     /// Get the gas costs
     pub const fn gas_costs(&self) -> &GasCosts {
         match self {
             Self::V1(params) => &params.gas_costs,
+            Self::V2(params) => &params.gas_costs,
         }
     }
 
     /// Get the base asset ID
     pub const fn base_asset_id(&self) -> &AssetId {
         match self {
             Self::V1(params) => &params.base_asset_id,
+            Self::V2(params) => &params.base_asset_id,
         }
     }
 
     /// Get the block gas limit
     pub const fn block_gas_limit(&self) -> u64 {
         match self {
             Self::V1(params) => params.block_gas_limit,
+            Self::V2(params) => params.block_gas_limit,
+        }
+    }
+
+    /// Get the block transaction size limit
+    pub fn block_transaction_size_limit(&self) -> u64 {
+        match self {
+            Self::V1(params) => params
+                .block_gas_limit
+                .checked_div(self.fee_params().gas_per_byte())
+                .unwrap_or(Self::DEFAULT_BLOCK_TRANSACTION_SIZE_LIMIT),
```

Comment:
After thinking a little bit more about it, I think we need to return `u64::MAX` for backward compatibility reasons. Because in the past it was not limited

Reply:
Updated in https://github.com/FuelLabs/fuel-vm/pull/821/commits/7b8c9985ccc3271a246064c93c47d71b9252d81d